### PR TITLE
Single run fix

### PR DESCRIPTION
**Pull request by fayebutler**
@currycoder 

Wasn't sure whether to put the try/except inside the run_schedule or just in the main.
Seems to work though!

### CODE REVIEW

File: scheduler.py [hunk 1]
```diff
@@ -216,7 +220,11 @@ def run_schedule(dryrun=False):
     args = parser.parse_args()
     error_list = Errors()
     
-    run_schedule(args.test)
+    try:
+        run_schedule(args.test)
```
Comment (currycoder) [inline scheduler.py]: It's probably worth moving the `singleton` call from `run_schedule` and in to the main function as you commented.  That way, we'd avoid accidentally picking up a `SystemExit` from somewhere else in `run_schedule`.  That may be unlikely, but generally it's good practice to have as little code running as possible within a try/except block.

File: scheduler.py [hunk 2]
```diff
@@ -154,6 +154,12 @@ def _run(dryrun=False):
...
     are scheduled to run now.
     """   
+    try:
+        me = singleton.SingleInstance()
+    except SystemExit:
+        print "** Quitting run at %s" % datetime.now().isoformat()   
+        sys.exit(-1)
+        
     run_logger = RunLogger()
     logging.config.dictConfig(LOGGING)
```
Comment (fayebutler) [inline scheduler.py]: @currycoder does in here work better? 
surrounding just the singleton line works as well.

Comment (currycoder) [inline scheduler.py]: That's the guy!

**Comment by currycoder** [comment]
Happy for this to be merged

**Pull request merged**